zio/ndjsonio/parser.go

```diff
@@ -206,6 +206,16 @@ func (p *Parser) jsonParseNumber(b []byte) (zng.Type, error) {
 }
 
 func (p *Parser) jsonParseString(b []byte) (zng.Type, error) {
-	p.builder.Append(zng.Unescape(b), false)
+	s, err := jsonparser.ParseString(b)
+	if err != nil {
+		return nil, err
+	}
+	p.builder.Append(zng.Unescape([]byte(s)), false)
```

Comment:
Avoid an allocation when b does not contain \\:
```go
b, err := jsonparser.Unescape(b, nil)
if err != nil {
	return nil, err
}
p.builder.Append(zng.Unescape(b), false)
```

Reply:
Ah, nice catch, updated.